### ISSUE TITLE
Not sure if `OpenFileDialog.Directory` is working properly

### ISSUE THREAD

**Issue by define-private-public** [AvaloniaUI/Avalonia#4141]
I'm running on Ubuntu Linux so it should be creating a GTK dialog under the hood.  I was trying to make a "open file" button which will launch an `OpenFileDialog`.  If there was a previously opened dialog, I wanted to to open in that directory.  It doesn't seem to be opening in the desired directory:

 Here's my code:

```
OpenFileDialog dialog = new OpenFileDialog()
{
    Directory = "/home/ben/Desktop/tmp/",
}
```

Instead of opening the file dialog in that `tmp` directory, it's opening up the `Desktop` directory with `tmp` as the pre-selected item.  Is this the desired behaviour (and I interpreted it wrong) or is this a bug?  Say instead if I use `InitialFilename="/home/ben/Desktop/tmp/song.ogg"`, then it is opening the dialog in the `tmp` folder.

**Comment by LiteracyFanatic** [comment]
I also find this behavior odd, especially because `OpenFolderDialog` does the same thing. In that case is doesn't make sense to provide an initial file. In fact, if the folder is empty, there is no way to open the dialog inside of it except for creating a temporary file and deleting it afterwards.

**Comment by timothyparez** [comment]
Experiencing the same issue.

If I set `Directory` to `/home/timothy/Pictures` or `/home/timothy/Pictures/` the dialog will open `/home/timothy`.  
If however I set it to `/home/timothy/Pictures/somerandomstring` where `somerandomstring` does not   
represent an existing file or folder it will open `/home/timothy/Pictures/`.

I do not believe this is the desired behavior. For example, the proper way to get the user's `Pictures` folder would be;

```
var picturesFolder = Environment.GetFolderPath(Environment.SpecialFolder.MyPictures);
```

This returns `/home/timothy/Pictures/` and if you then use it for the `OpenFileDialog` it opens
the home folder instead.
